src/html/fetch-external.js

```diff
@@ -0,0 +1,114 @@
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const crypto = require('crypto');
+const { setdefault } = require('ferrum');
+const { MountConfig } = require('@adobe/helix-shared');
+
+async function fetchExternal(context, { logger, request, downloadManager }) {
+  setdefault(context, 'content', {});
+  if (!request || !request.params) {
+    /* istanbul ignore next */
+    throw new Error('Request parameters missing');
+  }
+
+  // check for fstab
+  const fstabTask = downloadManager.getTaskById('fstab');
+  if (!fstabTask) {
+    logger.info('unable to fetch from external. no fstab task scheduled.');
+    return;
+  }
+
+  try {
+    const res = await fstabTask;
+    if (res.status !== 200) {
+      logger.info(`unable to fetch fstab.yaml: ${res.status}`);
+      return;
+    }
+    const cfg = await new MountConfig()
+      .withSource(res.body)
+      .init();
+
+    const idx = request.params.path.lastIndexOf('.');
+    const resourcePath = decodeURIComponent(request.params.path.substring(0, idx));
+    const mp = cfg.match(resourcePath);
+    if (!mp) {
+      logger.info(`no mount point for ${resourcePath}`);
+      return;
+    }
+
+    const url = new URL('https://adobeioruntime.net');
+    switch (mp.type) {
+      case 'google':
+        /* istanbul ignore next */
+        if (!mp.id) {
+          logger.warn('google docs mountpoint needs id');
+          return;
+        }
+        url.pathname = '/api/v1/web/helix/helix-services/gdocs2md@latest';
+        url.searchParams.append('path', mp.relPath);
+        url.searchParams.append('rootId', mp.id);
+        break;
+      case 'onedrive':
+        url.pathname = '/api/v1/web/helix/helix-services/word2md@latest';
+        url.searchParams.append('path', `${mp.relPath}.docx`);
+        url.searchParams.append('shareLink', mp.url);
+        break;
+      default:
+        if (mp.type) {
+          /* istanbul ignore next */
+          logger.info(`mount point type '${mp.type}' not supported.`);
+        } else {
+          logger.info(`mount point type not detected for '${mp.url}'.`);
+        }
+        return;
+    }
+
+    // extract some tracing info
+    const {
+      owner, repo, ref,
+    } = request.params;
+    const requestId = request.headers['x-request-id']
+      || request.headers['x-cdn-request-id']
+      || request.headers['x-openwhisk-activation-id']
+      || '';
+    url.searchParams.append('rid', requestId);
+    url.searchParams.append('src', `${owner}/${repo}/${ref}`);
+
+    const resp = await downloadManager.fetch({
+      uri: url.href,
+      // ump the timeout a bit, since the external script might take a while to render
+      options: {
+        timeout: 10000,
+      },
+      errorOn404: true,
+    });
+
+    // store extra source location if present
+    const sourceLocation = resp.headers['x-source-location'] || '';
```

Comment:
```suggestion
    const sourceLocation = resp.headers['x-source-location'] || '';
    context.sources = [...context.sources, sourceLocation || url.href ];
```

We should store the source, so that it can be included in the Surrogate-Keys header here: https://github.com/adobe/helix-pipeline/blob/87fa02657623338940060296565b8ed1f6ad2a81/src/html/set-surrogate-key.js